internal/cachecontroller/cache_controller.go

```diff
@@ -100,15 +100,22 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
+	_, present := c.inflightInvalidations.LoadOrStore(storeID, struct{}{})
+	if !present {
+		span.SetAttributes(attribute.Bool("check_invalidation", true))
+		// if the cache cannot be found, we want to invalidate entries in the background
+		// so that it does not block the answer path.
+		go func() {
+			// important to propagate the context without cancel to avoid
+			// cancelling the invalidation when parent request is complete.
+			c.findChangesAndInvalidate(context.WithoutCancel(ctx), storeID)
```

Comment:
Discussed with @elbuo8 offline, we want to have this span detached from the original trace to avoid increases trace length un-necessarily.